subsys/mgmt/mcumgr/transport/src/smp.c

```diff
@@ -155,6 +160,33 @@ int smp_transport_init(struct smp_transport *smpt)
 	return 0;
 }
 
+#ifdef CONFIG_SMP_CLIENT
+struct smp_transport *smp_client_transport_get(int smpt_type)
+{
+	struct smp_client_transport_entry *entry;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&smp_transport_clients, entry, node) {
+		if (entry->smpt_type == smpt_type) {
+			return entry->smpt;
+		}
+	}
+
+	return NULL;
+}
+
+void smp_client_transport_register(struct smp_client_transport_entry *entry)
+{
+	if (smp_client_transport_get(entry->smpt_type)) {
+		/* Already in list */
```

Comment:
We should probably have some warning here, as a LOG_DBG, as this would be strange behavior when certain transport type is registered twice; this also checks type only so it does not distinguish between different handler being registered for the same type.

Reply:
I can add that no problem.